src/rebase.c

```diff
@@ -270,8 +269,11 @@ static git_rebase *rebase_alloc(const git_rebase_options *rebase_opts)
 		git_rebase_init_options(&rebase->options, GIT_REBASE_OPTIONS_VERSION);
 
 	if (rebase_opts && rebase_opts->rewrite_notes_ref) {
-		if ((rebase->options.rewrite_notes_ref = git__strdup(rebase_opts->rewrite_notes_ref)) == NULL)
+		if ((rebase->options.rewrite_notes_ref =
+			git__strdup(rebase_opts->rewrite_notes_ref)) == NULL) {
+			git__free(rebase);
```

Comment:
Yes, I definitly agree, I've been reasoning about the same two options when I fixed this. I'd prefer to convert the function to return an `int` instead, as this is how nearly all of our functions which may fail operate. As such I deem it more idiomatic for this code base.